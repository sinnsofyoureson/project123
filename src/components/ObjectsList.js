import React from 'react'
import PropTypes from 'prop-types'
import SingleObject from './SingleObject.js'

const ObjectsList = ({ objects, onObjectClick, onShowInfo }) => (
    <ul style={{
        maxHeight: '65vh',
        width: '400px',
        background: '#080808',
        borderRadius: '8px',
        boxShadow: '0 8px 21px #000',
        padding: '0',
        margin: '34px',
        overflowY: 'scroll'
    }}>
        {objects.map(object => 
            <SingleObject
                key={object.id}
                {...object}
                onClick={() => onObjectClick(object.id)}
                showObjectInfo={() => onShowInfo(object.id)} />
        )}
    </ul>
)

ObjectsList.propTypes = {
    objects: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
        address: PropTypes.string.isRequired,
        client: PropTypes.string.isRequired,
        contract: PropTypes.string.isRequired,
        services: PropTypes.string,
        status: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired,
        expiration: PropTypes.string,
        active: PropTypes.bool.isRequired,
        shown: PropTypes.bool.isRequired
    }).isRequired).isRequired,
    onObjectClick: PropTypes.func.isRequired,
    onShowInfo: PropTypes.func.isRequired
}

export default ObjectsList