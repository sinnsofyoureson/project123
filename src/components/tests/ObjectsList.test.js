import React from 'react';
import renderer from 'react-test-renderer';

// components and containers testing
import ObjectsList from '../ObjectsList';

describe('ObjectsList component', () => {
    it('matches the snapshot', () => {
      const props = {
        objects: []
      }
      const tree = renderer.create(<ObjectsList {...props} />).toJSON();
      jestExpect(tree).toMatchSnapshot();
    })
});