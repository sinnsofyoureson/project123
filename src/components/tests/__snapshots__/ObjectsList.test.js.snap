// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectsList component matches the snapshot 1`] = `
<ul
  style={
    Object {
      "background": "#080808",
      "borderRadius": "8px",
      "boxShadow": "0 8px 21px #000",
      "margin": "34px",
      "maxHeight": "65vh",
      "overflowY": "scroll",
      "padding": "0",
      "width": "400px",
    }
  }
/>
`;
